org.eclipse.dawnsci.nexus/src/org/eclipse/dawnsci/nexus/NexusConstants.java

```diff
@@ -0,0 +1,143 @@
+	public final static String INSTRUMENT = "NXinstrument";
+	public final static String LOG = "NXlog";
+	public final static String MIRROR = "NXmirror";
+	public final static String MODERATOR = "NXmoderator";
+	public final static String MONITOR = "NXmonitor";
+	public final static String MONOCHROMATOR = "NXmonochromator";
+	public final static String NOTE = "NXnote";
+	public final static String OBJECT = "NXobject";
+	public final static String ORIENTATION = "NXorientation";
+	public final static String PARAMETERS = "NXparameters";
+	public final static String PINHOLE = "NXpinhole";
+	public final static String POLARIZER = "NXpolarizer";
+	public final static String POSITIONER = "NXpositioner";
+	public final static String PROCESS = "NXprocess";
+	public final static String REFLECTIONS = "NXreflections";
+	public final static String ROOT = "NXroot";
+	public final static String SAMPLECOMPONENT = "NXsample_component";
+	public final static String SAMPLE = "NXsample";
+	public final static String SENSOR = "NXsensor";
+	public final static String SHAPE = "NXshape";
+	public final static String SLIT = "NXslit";
+	public final static String SOURCE = "NXsource";
+	public final static String SUBENTRY = "NXsubentry";
+	public final static String TRANSFORMATIONS = "NXtransformations";
+	public final static String TRANSLATION = "NXtranslation";
+	public final static String USER = "NXuser";
+	public final static String VELOCITYSELECTOR = "NXvelocity_selector";
+	public final static String XRAYLENS = "NXxraylens";
+
+	/**
+	 * NXdata default data name
+	 */
+	public static final String DATA_DATA = "data";
+	/**
+	 * NXdata dataset attribute
+	 */
+	public static final String DATA_NAME = "long_name";
+	/**
+	 * NXdata attribute that points to main dataset
+	 */
+	public static final String DATA_SIGNAL = "signal";
+	/**
+	 * NXdata attribute that maps dimensions of main dataset to axis datasets
+	 */
+	public static final String DATA_AXES = "axes";
+	/**
+	 * NXdata attribute that points to error datasets (may also be legacy if attached to main dataset)
+	 */
+	public static final String DATA_UNCERTAINTIES ="uncertainties";
+
+	/**
+	 * NXdata indicate no default for axis in axes attribute
+	 */
+	public static final String DATA_AXESEMPTY = ".";
+
+	/**
+	 * Suffix for name of error datasets
+	 */
+	public static final String DATA_ERRORS_SUFFIX = "_errors";
+	/**
+	 * Suffix for name of attributes that indicate dimension mappings from axis dataset to main dataset
+	 */
+	public static final String DATA_INDICES_SUFFIX = "_indices";
+	/**
+	 * Suffix for names of datasets
+	 */
+	public static final String DATA_UNCERTAINTY_SUFFIX ="_uncertainty";
+	/**
+	 * Suffix for names of datasets that contain set values (for axes)
+	 */
+	public static final String DATA_AXESSET_SUFFIX = "_set";
+
+	/**
+	 * legacy attribute name of main dataset
+	 */
+	@Deprecated
+	public static final String DATA_ERRORS = "errors";
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)